ui/components/src/LineChart/LineChart.tsx

```diff
@@ -205,20 +209,56 @@ export function LineChart({
       onClick={(e) => {
         // Pin and unpin when clicking on chart canvas but not tooltip text.
         if (e.target instanceof HTMLCanvasElement) {
-          setIsTooltipPinned((current) => !current);
+          setTooltipPinnedCoords((current) => {
+            if (current === null) {
+              return {
+                page: {
+                  x: e.pageX,
+                  y: e.pageY,
+                },
+                client: {
+                  x: e.clientX,
+                  y: e.clientY,
+                },
+                plotCanvas: {
+                  x: e.nativeEvent.offsetX,
+                  y: e.nativeEvent.offsetY,
+                },
+                target: e.target,
+              };
+            } else {
+              return null;
+            }
+          });
         }
       }}
       onMouseDown={(e) => {
-        // Hide tooltip when user drags to zoom, but allow clicking inside tooltip to copy labels.
-        if (e.target instanceof HTMLCanvasElement) {
-          setShowTooltip(false);
+        e.preventDefault();
+        const { clientX } = e;
+        setDragging(true);
+        setStartX(clientX);
+      }}
+      onMouseMove={(e) => {
+        const { clientX } = e;
+        if (isDragging) {
```

Comment:
These lines fix the "tooltip disappears and reappears" bug @juliepagano mentioned [here](https://github.com/perses/perses/pull/1181#issuecomment-1558087880) in #1181. Before `setShowTooltip(false);` was being called while the tooltip was being pinned, I've added a useState to confirm whether the user is clicking and dragging their cursor to zoom before calling  `setShowTooltip(false);` (since you don't want the tooltip to get in the way while trying to select the range). 

Also note, the ECharts [datazoom](https://echarts.apache.org/en/api.html#events.datazoom) event above fires after the data range is selected, there may be other built-in events/actions we can leverage but doing it purely in React seems to work pretty well 🤷‍♂️